src/imp/security_framework.rs

```diff
@@ -351,6 +359,24 @@ impl TlsAcceptor {
     }
 }
 
+pub struct ChainIterator<'a, S: 'a> {
+    trust: Option<SecTrust>,
+    pos: usize,
+    _stream: &'a TlsStream<S>,
+}
+impl<'a, S> Iterator for ChainIterator<'a, S> {
+    type Item = Certificate;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        if let Some(trust) = self.trust.as_ref() {
+            let pos = self.pos;
+            self.pos += 1;
+            return trust.certificate_at_index(pos as _).map(Certificate);
```

Comment:
This needs to stop at the end of the chain, right?

Reply:
Trust will return None once it is over limit.